<a href="https://colab.research.google.com/github/switlanakostyuk-ctrl/Apollo/blob/main/%22hw_9_1_%D0%9E%D0%BF%D0%B5%D1%80%D0%B0%D1%86%D1%96%D1%97_%D0%BD%D0%B0%D0%B4_%D0%B4%D0%B0%D0%BD%D0%B8%D0%BC%D0%B8_ipynb%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402
result = (a+b)*c/d
result

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [3]:
remainder = 11283 % 89
remainder

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [4]:
integer_division = 11283 // 83
integer_division

135

4. Запишіть у змінну `name` ваше імʼя.

In [6]:
name = "Svitlana"

5. Запишіть у змінну `surname` ваше прізвище.

In [7]:
surname = "Kostiuk"

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [8]:
name_surname = name + "  " + surname
print(name_surname)

Svitlana  Kostiuk


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [9]:
name_surname_upper = name_surname.upper()
name_surname_upper

'SVITLANA  KOSTIUK'

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [12]:
name_surname[2]

'i'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [13]:
(surname[0], surname[-1])

('K', 'k')

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [14]:
len(surname)

7

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [17]:
city = "Antananarivo"
city.lower().count("a")

4

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [19]:
my_string = "The quick brown fox jumps over the lazy dog!!!"

print(my_string.startswith("The"))
print(my_string.endswith("dog!"))

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [20]:
words = my_string.split ()
result_string = "_".join(words)
print(result_string)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`